### ISSUE TITLE
`SQLParseException` error during `CREATE TABLE` with a generated column and `CASE` statement

### ISSUE THREAD

**Issue by hammerhead** [crate/crate#16953]
### CrateDB version

5.9.2

### CrateDB setup information

Standard single node

### Problem description

Creating a table with a generated column that uses a `CASE` expression fails with an unexpected syntax error. The error manifests in two different ways, depending on whether the table is partitioned or not.

**Not partitioned**: The error is shown right away when submitting the `CREATE TABLE` statement
**Partitioned**: `CREATE TABLE` is accepted, but any subsequent queries fail. Also against unrelated system tables.

### Steps to Reproduce

***Not partitioned***

```sql
CREATE TABLE q (
  a INTEGER,
  b INTEGER GENERATED ALWAYS AS CASE WHEN a > 0 THEN 5 ELSE NULL END
);
```

***Partitioned***

```sql
CREATE TABLE q (
  a INTEGER,
  b INTEGER GENERATED ALWAYS AS CASE WHEN a > 0 THEN 5 ELSE NULL END
)
PARTITIONED BY (a);
```

### Actual Result

***Not partitioned***
`CREATE TABLE` fails right away with:
```
io.crate.exceptions.SQLParseException: line 1:10: no viable alternative at input '(true,'
	at io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
	at io.crate.exceptions.SQLExceptions.prepareForClientTransmission(SQLExceptions.java:200)
	at io.crate.rest.action.SqlHttpHandler.sendResponse(SqlHttpHandler.java:172)
	at io.crate.rest.action.SqlHttpHandler.lambda$channelRead0$0(SqlHttpHandler.java:127)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
	at io.crate.rest.action.RestRowCountReceiver.fail(RestRowCountReceiver.java:84)
	at io.crate.session.RowConsumerToResultReceiver.accept(RowConsumerToResultReceiver.java:63)
	at io.crate.planner.node.ddl.CreateTablePlan.lambda$executeOrFail$0(CreateTablePlan.java:76)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
	at io.crate.action.FutureActionListener.onFailure(FutureActionListener.java:40)
	at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:173)
	at org.elasticsearch.action.ActionListener$7.onFailure(ActionListener.java:294)
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:106)
	at org.elasticsearch.action.ActionListener.accept(ActionListener.java:55)
	at org.elasticsearch.action.ActionListener.accept(ActionListener.java:38)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
	at org.elasticsearch.cluster.AckedClusterStateUpdateTask.onFailure(AckedClusterStateUpdateTask.java:96)
	at org.elasticsearch.cluster.service.MasterService$SafeClusterStateTaskListener.onFailure(MasterService.java:477)
	at org.elasticsearch.cluster.service.MasterService$TaskOutputs.notifyFailedTasks(MasterService.java:404)
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:176)
	at org.elasticsearch.cluster.service.MasterService.lambda$doStart$0(MasterService.java:121)
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:170)
	at org.elasticsearch.cluster.service.BatchedTask.run(BatchedTask.java:70)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:231)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:194)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: io.crate.sql.parser.ParsingException: line 1:10: no viable alternative at input '(true,'
	at io.crate.sql.parser.SqlParser$1.syntaxError(SqlParser.java:58)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportNoViableAlternative(DefaultErrorStrategy.java:310)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:136)
	at io.crate.sql.parser.antlr.SqlBaseParser.primaryExpression(SqlBaseParser.java:11345)
	at io.crate.sql.parser.antlr.SqlBaseParser.valueExpression(SqlBaseParser.java:10444)
	at io.crate.sql.parser.antlr.SqlBaseParser.predicated(SqlBaseParser.java:9714)
	at io.crate.sql.parser.antlr.SqlBaseParser.booleanExpression(SqlBaseParser.java:9563)
	at io.crate.sql.parser.antlr.SqlBaseParser.expr(SqlBaseParser.java:9182)
	at io.crate.sql.parser.antlr.SqlBaseParser.explicitFunction(SqlBaseParser.java:12059)
	at io.crate.sql.parser.antlr.SqlBaseParser.primaryExpression(SqlBaseParser.java:11006)
	at io.crate.sql.parser.antlr.SqlBaseParser.valueExpression(SqlBaseParser.java:10444)
	at io.crate.sql.parser.antlr.SqlBaseParser.predicated(SqlBaseParser.java:9714)
	at io.crate.sql.parser.antlr.SqlBaseParser.booleanExpression(SqlBaseParser.java:9563)
	at io.crate.sql.parser.antlr.SqlBaseParser.expr(SqlBaseParser.java:9182)
	at io.crate.sql.parser.antlr.SqlBaseParser.singleExpression(SqlBaseParser.java:488)
	at io.crate.sql.parser.SqlParser.invokeParser(SqlParser.java:128)
	at io.crate.sql.parser.SqlParser.generateExpression(SqlParser.java:103)
	at io.crate.sql.parser.SqlParser.createExpression(SqlParser.java:99)
	at io.crate.metadata.doc.DocTableInfoFactory.create(DocTableInfoFactory.java:294)
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$add$4(MetadataCreateIndexService.java:725)
	at org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:424)
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.add(MetadataCreateIndexService.java:711)
	at io.crate.execution.ddl.tables.TransportCreateTableAction$1.execute(TransportCreateTableAction.java:180)
	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45)
	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:665)
	at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:280)
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:175)
	... 8 more
Caused by: org.antlr.v4.runtime.NoViableAltException
	at org.antlr.v4.runtime.atn.ParserATNSimulator.noViableAlt(ParserATNSimulator.java:2014)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.execATN(ParserATNSimulator.java:445)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.adaptivePredict(ParserATNSimulator.java:371)
	at io.crate.sql.parser.antlr.SqlBaseParser.primaryExpression(SqlBaseParser.java:10989)
	... 31 more
```

***Partitioned***
`CREATE TABLE` succeeds, but queries such as `SELECT * FROM information_schema.tables` fail:

```
io.crate.exceptions.SQLParseException: line 1:10: no viable alternative at input '(true,'
	at io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
	at io.crate.exceptions.SQLExceptions.prepareForClientTransmission(SQLExceptions.java:200)
	at io.crate.rest.action.SqlHttpHandler.sendResponse(SqlHttpHandler.java:172)
	at io.crate.rest.action.SqlHttpHandler.lambda$channelRead0$0(SqlHttpHandler.java:127)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
	at io.crate.rest.action.RestResultSetReceiver.fail(RestResultSetReceiver.java:88)
	at io.crate.session.RetryOnFailureResultReceiver.fail(RetryOnFailureResultReceiver.java:98)
	at io.crate.session.RowConsumerToResultReceiver.consumeIt(RowConsumerToResultReceiver.java:114)
	at io.crate.session.RowConsumerToResultReceiver.accept(RowConsumerToResultReceiver.java:57)
	at io.crate.execution.engine.InterceptingRowConsumer.lambda$tryForwardResult$1(InterceptingRowConsumer.java:93)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: io.crate.sql.parser.ParsingException: line 1:10: no viable alternative at input '(true,'
	at io.crate.sql.parser.SqlParser$1.syntaxError(SqlParser.java:58)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportNoViableAlternative(DefaultErrorStrategy.java:310)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:136)
	at io.crate.sql.parser.antlr.SqlBaseParser.primaryExpression(SqlBaseParser.java:11345)
	at io.crate.sql.parser.antlr.SqlBaseParser.valueExpression(SqlBaseParser.java:10444)
	at io.crate.sql.parser.antlr.SqlBaseParser.predicated(SqlBaseParser.java:9714)
	at io.crate.sql.parser.antlr.SqlBaseParser.booleanExpression(SqlBaseParser.java:9563)
	at io.crate.sql.parser.antlr.SqlBaseParser.expr(SqlBaseParser.java:9182)
	at io.crate.sql.parser.antlr.SqlBaseParser.explicitFunction(SqlBaseParser.java:12059)
	at io.crate.sql.parser.antlr.SqlBaseParser.primaryExpression(SqlBaseParser.java:11006)
	at io.crate.sql.parser.antlr.SqlBaseParser.valueExpression(SqlBaseParser.java:10444)
	at io.crate.sql.parser.antlr.SqlBaseParser.predicated(SqlBaseParser.java:9714)
	at io.crate.sql.parser.antlr.SqlBaseParser.booleanExpression(SqlBaseParser.java:9563)
	at io.crate.sql.parser.antlr.SqlBaseParser.expr(SqlBaseParser.java:9182)
	at io.crate.sql.parser.antlr.SqlBaseParser.singleExpression(SqlBaseParser.java:488)
	at io.crate.sql.parser.SqlParser.invokeParser(SqlParser.java:128)
	at io.crate.sql.parser.SqlParser.generateExpression(SqlParser.java:103)
	at io.crate.sql.parser.SqlParser.createExpression(SqlParser.java:99)
	at io.crate.metadata.doc.DocTableInfoFactory.create(DocTableInfoFactory.java:294)
	at io.crate.metadata.doc.DocSchemaInfo.lambda$getTableInfo$1(DocSchemaInfo.java:153)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1710)
	at io.crate.metadata.doc.DocSchemaInfo.getTableInfo(DocSchemaInfo.java:151)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212)
	at java.base/java.util.HashMap$KeySpliterator.tryAdvance(HashMap.java:1736)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:161)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1949)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:295)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1950)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
	at java.base/java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:720)
	at java.base/java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:720)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1949)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1949)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at io.crate.data.CollectingBatchIterator.moveNext(CollectingBatchIterator.java:129)
	at io.crate.data.MappedForwardingBatchIterator.moveNext(MappedForwardingBatchIterator.java:41)
	at io.crate.data.LimitingBatchIterator.moveNext(LimitingBatchIterator.java:57)
	at io.crate.session.RowConsumerToResultReceiver.consumeIt(RowConsumerToResultReceiver.java:75)
	... 5 more
Caused by: org.antlr.v4.runtime.NoViableAltException
	at org.antlr.v4.runtime.atn.ParserATNSimulator.noViableAlt(ParserATNSimulator.java:2014)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.execATN(ParserATNSimulator.java:445)
	at org.antlr.v4.runtime.atn.ParserATNSimulator.adaptivePredict(ParserATNSimulator.java:371)
	at io.crate.sql.parser.antlr.SqlBaseParser.primaryExpression(SqlBaseParser.java:10989)
	... 56 more
```

### Expected Result

***Not partitioned***
Creating the table succeeds.

***Partitioned***
Creating the table succeeds, but also rejects any errors during table creation instead of later breaking queries on system tables.

**Comment by wierdvanderhaar** [comment]
For a prospect, we have this situation: 
With the table created the cluster becomes red and every option trying to get rid of the table results in:

`SQLParseException[line 1:10: no viable alternative at input '(true,']`

**Comment by seut** [comment]
@hammerhead Thank you for reporting. We've fixed this by #16976, it will be released with the next hotfix release.

**Comment by wierdvanderhaar** [comment]
Because this: 
[bug](https://github.com/crate/crate/issues/16953)

The cluster was broken. As waiting for a new release takes too much time and the data can be reloaded easily using the Argo pipelines, we agreed on recreating the CrateDB Cluster (K8s) and reloading the data.  

Cluster is green again, and Anthony gave an alternative for the eot_bis table using generated columns. 

[example](https://github.com/crate/pre-sales/blob/2c91cab81636b6a7d428b70373cbaff78a0d843c/customers/SMEG/Example-eot_bis.sql)
